web/src/components/storage/ProposalVolumes.jsx

```diff
@@ -203,25 +206,48 @@ const VolumeRow = ({ columns, volume, options, isLoading, onEdit, onDelete }) =>
     );
   };
 
-  const Details = ({ volume, options }) => {
+  const Details = ({ volume }) => {
     const snapshots = hasSnapshots(volume);
     const transactional = isTransactionalRoot(volume);
 
-    // TRANSLATORS: the filesystem uses a logical volume (LVM)
-    const text = `${volume.fsType} ${options.lvm ? _("logical volume") : _("partition")}`;
-    const lockIcon = <Icon name="lock" size="xxxs" />;
-    const snapshotsIcon = <Icon name="add_a_photo" size="xxxs" />;
-    const transactionalIcon = <Icon name="sync" size="xxxs" />;
+    const text = () => {
+      if (volume.target === "filesystem")
+        // TRANSLATORS: %s will be replaced by a file-system type like "Btrfs" or "Ext4"
+        return sprintf(_("Reused %s"), volume.targetDevice?.filesystem?.type || "");
+      if (transactional)
+        return _("Transactional Btrfs");
+      if (snapshots)
+        return _("Btrfs with snapshots");
+
+      return volume.fsType;
+    };
+
+    return (
+      <div className="split">
+        <span>{text()}</span>
```

Comment:
I would not use this kind of function. As far as I can see, you can use a `let text;` and then fill it's value conditionally and finally use it directly.

```js
let text;

if (volume.target === "filesystem") {
  // TRANSLATORS: %s will be replaced by a file-system type like "Btrfs" or "Ext4"
  text =  sprintf(_("Reused %s"), volume.targetDevice?.filesystem?.type || "");
} else if (transactional) {
   text =  _("Transactional Btrfs");
} else if (snapshots) {
   text = _("Btrfs with snapshots");
};
```

Or whatever other technique but not a definition and evaluation of a function.

Reply:
The above is because I expect this internal component to be re-rendered as soon as its volume prop changes.